### ISSUE TITLE
`calcite-tile` with `href` is not keyboard accessible via Enter key

### ISSUE THREAD

**Issue by gabrielegiunchi** [Esri/calcite-design-system#14557]
**monday.com sync:** #12180058664

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

N/A, Tile

### Summary

When a `calcite-tile` is provided with an `href` property, it behaves visually as a link but cannot be activated using the keyboard.

### Actual Behavior

Nothing happens when Enter is pressed while the focused tile has an `href`.

### Expected Behavior

The tile should activate the link and navigate to the target URL when Enter is pressed

### Reproduction Sample

https://codepen.io/DintaMel-the-lessful/pen/raWEMBa

### Reproduction Steps

1. Render a `calcite-tile` with an `href` property.
2. Navigate to the tile using the keyboard (Tab key).
3. Press Enter while the tile is focused.

### Reproduction Version

5.1.0

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Scene Viewer

**Comment by DintaMel** [comment]
I updated the reproduction codepen so that is links to an `https` site, since codepen blocks navigation.

**Comment by isaacbraun** [comment]
Verified on `5.2.0-next.71` via https://codepen.io/isaacbraun/pen/pvePRwp that the Tile navigates to the linked `href` when triggered via the `enter` keypress. Also tested that selecting when within Tile Group works as expected.
